ops/testing.py

```diff
@@ -182,9 +196,10 @@ def add_relation(self, relation_name, remote_app):
             self._model.relations._invalidate(relation_name)
         if self._charm is None or not self._hooks_enabled:
             return rel_id
+        # TODO: Once we land relation_created support, it should be triggered by this code.
         return rel_id
 
-    def add_relation_unit(self, relation_id, remote_unit_name):
+    def add_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
```

Comment:
Just an example of above "compactness" exposed here, to spark discussion:
```
    def add_relation_unit(self, relation_id:int, remote_unit_name:str) -> None:
```

Reply:
I'm using 'autopep8' in my editor, which is adding the whitespace. Just to confirm I tried your version and asked flake8:
```
$ flake8
./ops/testing.py:202:44: E231 missing whitespace after ':'
./ops/testing.py:202:66: E231 missing whitespace after ':'
```

It seems unless we want to tweak our flake8 config, we can't go more compact.